mh/quest_utils.py

```diff
@@ -1402,10 +1413,22 @@ def make_binary_event_quest(quest_data):
     data += b'\0' * (4 * 11)
 
     assert len(data) == 0x4B4
-    try:
-        data += read_quest_sm_data(quest_info['smallmonster_data_file'])
-    except Exception as e:
-        pass
+
+    # Add 2-byte tail between the main quest info and the small monster data that
+    # describes how many bytes total the quest takes up before the optional
+    # arena equipment data
+    tail = 0x4B4 + 4
+    sm_data = read_quest_sm_data(quest_info['smallmonster_data_file'])
+    if sm_data is not None:
+        tail += len(sm_data) - 4
+        data += struct.pack(">I", tail)
+        data += sm_data[4:]
+    else:
+        data += struct.pack(">I", tail)
+
+    # If there is arena equipment data, add it to the end of the quest binary
+    if 'arena_equipment' in quest_data:
+        data += create_arena_equipment_set(quest_data['arena_equipment'])
```

Comment:
Added that line and pushed, since I can't think of a better place to put it without adding "is_arena" properties or something (which would be redundant information)